execution/execution-correctness/src/thread.rs

```diff
@@ -22,12 +23,13 @@ pub struct ThreadService {
 }
 
 impl ThreadService {
-    pub fn new(storage_addr: SocketAddr) -> Self {
+    pub fn new(storage_addr: SocketAddr, prikey: Ed25519PrivateKey) -> Self {
```

Comment:
every implementation of LEC should take in a secure storage where it obtains its private key
if we want to move forward with this PR, we should figure out 1) will be fix that in this PR and 2) if we do not, can we avoid making LEC a requirement while we fix it. Otherwise this will make for a very wonky implementation.

Note, I started from the bottom and am moving up today, so I don't know exactly what was done to get here.

Reply:
I don't quite understand you insist on taking secure storage into LEC service.